version: "3.5"
networks:
  dncore_network:
    name: dncore_network
    external: true
volumes:
  vpndnpdappnodeeth_data: {}
  vpndnpdappnodeeth_config: {}
services:
  vpn.dnp.dappnode.eth:
    build: .
    image: "vpn.dnp.dappnode.eth:0.2.8"
    container_name: DAppNodeCore-vpn.dnp.dappnode.eth
    privileged: true
    restart: unless-stopped
    volumes:
      - "/var/run/docker.sock:/var/run/docker.sock"
      - "/etc/hostname:/etc/vpnname:ro"
      - "/usr/src/dappnode/config:/usr/src/app/config:ro"
      - "/lib/modules:/lib/modules:ro"
      - "vpndnpdappnodeeth_data:/usr/src/app/secrets"
      - "vpndnpdappnodeeth_config:/etc/openvpn"
    ports:
      - "1194:1194/udp"
      - "8092:8092"
    dns: 172.33.1.2
    networks:
      dncore_network:
        ipv4_address: 172.33.1.4
        aliases:
          - "vpn.dnp.dappnode.eth.vpn.dappnode"
          - vpn.dappnode
    logging:
      driver: journald
